### ISSUE TITLE
`basic_parse` bug with unary minus sign

### ISSUE THREAD

**Issue by spencerlyon2** [symengine/symengine#1043]
Strings like `"foo* - bar"` will cause `basic_parse` to fail with the following message:

```
libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: Operator inconsistency!
```

I can try to handle this next week, unless someone beats me to it first 😏

**Comment by certik** [comment]
Thanks for reporting it. It looks like it is not supported yet. Shouldn't there be parentheses anyway around `-bar`?

**Comment by spencerlyon2** [comment]
The current parser expects there to be parenthesis, but they wouldn't be required.

Here's an example from sympy:

```
In [1]: import sympy

In [2]: sympy.sympify("foo * -bar")
Out[2]: -bar*foo
```

**Comment by Sumith1896** [comment]
cc @srajangarg

**Comment by spencerlyon2** [comment]
Works on master now

**Comment by certik** [comment]
This was fixed by #1046.
